File: src/test/java/com/pharmacy/utils/EncryptionUtilsTest.java
package com.pharmacy.utils;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class EncryptionUtilsTest {

    @Test
    void generateKeyBytes() {
        EncryptionUtils encryptionUtils = new EncryptionUtils();
        byte[] keyBytes = encryptionUtils.generateKeyBytes();

        assertEquals(16, keyBytes.length);
    }

    @Test
    void encrypt() {
        EncryptionUtils encryptionUtils = new EncryptionUtils();
        byte[] secretKey = encryptionUtils.generateKeyBytes();

        String message = "admin";
        byte[] cipherText = encryptionUtils.encrypt(message, secretKey);

        assertEquals(33, cipherText.length);
    }

    @Test
    void decrypt() {
        EncryptionUtils encryptionUtils = new EncryptionUtils();
        byte[] secretKey = encryptionUtils.generateKeyBytes();

        String message = "admin";
        byte[] cipherText = encryptionUtils.encrypt(message, secretKey);
        String decrypted = encryptionUtils.decrypt(cipherText, secretKey);

        assertEquals(message, decrypted);
    }
}

File: src/test/java/com/pharmacy/utils/EncryptionUtilsTest.kt
package com.pharmacy.utils

import kotlin.test.assertEquals
import org.junit.jupiter.api.Test

internal class EncryptionUtilsTest {
    @Test
    fun `test key bytes generation`() {
        val encryptionUtils = EncryptionUtils()
        val keyBytes = encryptionUtils.generateKeyBytes()

        assertEquals(16, keyBytes.size)
    }

    @Test
    fun `test encryption`() {
        val encryptionUtils = EncryptionUtils()
        val secretKey = encryptionUtils.generateKeyBytes()

        val message = "admin"
        val cipherText = encryptionUtils.encrypt(message, secretKey)

        assertEquals(33, cipherText.size)
    }

    @Test
    fun `test decryption`() {
        val encryptionUtils = EncryptionUtils()
        val secretKey = encryptionUtils.generateKeyBytes()

        val message = "admin"
        val cipherText = encryptionUtils.encrypt(message, secretKey)
        val decrypted = encryptionUtils.decrypt(cipherText, secretKey)

        assertEquals(message, decrypted)
    }
}
